src/kagglehub/http_resolver.py

```diff
@@ -199,6 +200,66 @@ def _inner_download_file(file: str) -> None:
         return out_path
 
 
+class NotebookOutputHttpResolver(Resolver[CodeHandle]):
+    def is_supported(self, *_, **__) -> bool:  # noqa: ANN002, ANN003
+        # Downloading files over HTTP is supported in all environments for all handles / paths.
+        return True
+
+    def __call__(self, h: CodeHandle, path: Optional[str] = None, *, force_download: Optional[bool] = False) -> str:
+        api_client = KaggleApiV1Client()
+
+        cached_response = load_from_cache(h, path)
+        if cached_response and not force_download:
+            return cached_response  # Already cached
+        elif cached_response and force_download:
+            delete_from_cache(h, path)
+
+        api_path = f"kernels/{h.owner}/{h.notebook}/output/download"
+        output_root = Path(get_cached_path(h, path))
+
+        # Create the intermediary directories
+        # List the files and decide how to download them:
+        # - <= 25 files: Download files in parallel
+        # > 25 files: Download the archive and uncompress
+        (files, has_more) = self._list_files(api_client, h)
+        if has_more:
+            # TODO(b/379761520): support .tar.gz archived downloads
+            logger.warning(f"Too many files in {h}. Unable to download files.")
+            return ""
+
+        # Download files individually in parallel
+        def _inner_download_file(file: str) -> None:
+            output_filepath = output_root / file
+            os.makedirs(os.path.dirname(output_filepath), exist_ok=True)
+            api_client.download_file(api_path + "/" + file, str(output_filepath), h)
+
+        thread_map(
+            _inner_download_file,
+            files,
+            desc=f"Downloading {len(files)} files",
+            max_workers=8,  # Never use more than 8 threads in parallel to download files.
+        )
+
+        mark_as_complete(h, path)
+
+        # TODO(b/377510971): when notebook is a Kaggle utility script, update sys.path
```

Comment:
When it is a utility script, the source and not the output is used. Is the name `notebook_output_download` still the right name?

Reply:
but maybe cc @djherbis @calderjo to confirm

Reply:
We do convert ipynb to a py and place it in both the source and output directories ([link](https://source.corp.google.com/h/kaggle-internal/codesearch/+/master:kaggle/Kaggle.Services.Kernels/Clients/SessionSchedulerClient.cs;l=434-455?q=%22utility%22%20%22scripts%22%20%22output%22&ss=h%2Fkaggle-internal%2Fcodesearch%2F%2B%2Frefs%2Fheads%2Fmaster:kaggle%2FKaggle.Services.Kernels%2F))

Reply:
Great. Thanks for checking.